superset-frontend/src/common/components/Fade.tsx

```diff
@@ -0,0 +1,28 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { styled } from '@superset-ui/core';
+
+type FadeProps = { hovered: boolean };
+
+const Fade = styled.div<FadeProps>`
+  transition: all ${({ theme }) => theme.transitionTiming}s;
+  opacity: ${props => (props.hovered ? 1 : 0)};
+`;
```

Comment:
This works... but would there be a downside to a simpler approach of just making it:
```
  transition: opacity ${({ theme }) => theme.transitionTiming}s;
  opacity: 0;
  &:hover {
    opacity: 1
  }
```
then, I _think_ you could go through and remove all the code regarding the `hovered` state and prop.

Reply:
With this approach **<Fade>** component would only fade in/out if we hover over the header which is not the previous behaviour we had. The previous behaviour would trigger fade in/out when we hover over **TableElement** component (header and body).